substrate/frame/staking/src/asset.rs

```diff
@@ -0,0 +1,125 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains all the interactions with [`Config::Currency`] to manipulate the underlying staking
+//! asset.
+
+use frame_support::traits::{Currency, InspectLockableCurrency, LockableCurrency};
+
+use crate::{BalanceOf, Config, NegativeImbalanceOf, PositiveImbalanceOf};
+
+/// Existential deposit for the chain.
+pub fn existential_deposit<T: Config>() -> BalanceOf<T> {
+	T::Currency::minimum_balance()
+}
+
+/// Total issuance of the chain.
+pub fn total_issuance<T: Config>() -> BalanceOf<T> {
+	T::Currency::total_issuance()
+}
+
+/// Total balance of `who`. Includes both, free and reserved.
+pub fn total_balance<T: Config>(who: &T::AccountId) -> BalanceOf<T> {
+	T::Currency::total_balance(who)
+}
+
+/// Stakeable balance of `who`.
+///
+/// This includes balance free to stake along with any balance that is already staked.
+pub fn stakeable_balance<T: Config>(who: &T::AccountId) -> BalanceOf<T> {
+	T::Currency::free_balance(who)
+}
+
+/// Balance of `who` that is currently at stake.
+///
+/// The staked amount is locked and cannot be transferred out of `who`s account.
+pub fn staked<T: Config>(who: &T::AccountId) -> BalanceOf<T> {
+	T::Currency::balance_locked(crate::STAKING_ID, who)
+}
+
+/// Set balance that can be staked for `who`.
+///
+/// This includes any balance that is already staked.
+#[cfg(any(test, feature = "runtime-benchmarks"))]
+pub fn set_stakeable_balance<T: Config>(who: &T::AccountId, value: BalanceOf<T>) {
+	T::Currency::make_free_balance_be(who, value);
+}
+
+/// Update `amount` at stake for `who`.
+///
+/// Overwrites the existing stake amount. If passed amount is lower than the existing stake, the
+/// difference is unlocked.
+pub fn update_stake<T: Config>(who: &T::AccountId, amount: BalanceOf<T>) {
+	T::Currency::set_lock(
+		crate::STAKING_ID,
+		who,
+		amount,
+		frame_support::traits::WithdrawReasons::all(),
+	);
+}
+
+/// Kill the stake of `who`.
+///
+/// All locked amount is unlocked.
+pub fn kill_stake<T: Config>(who: &T::AccountId) {
+	T::Currency::remove_lock(crate::STAKING_ID, who);
+}
+
+/// Slash the value from `who`.
+///
+/// A negative imbalance is returned which can be resolved to deposit the slashed value.
+pub fn slash<T: Config>(
+	who: &T::AccountId,
+	value: BalanceOf<T>,
+) -> (NegativeImbalanceOf<T>, BalanceOf<T>) {
+	T::Currency::slash(who, value)
+}
+
+/// Mint `value` into an existing account `who`.
+///
+/// This does not increase the total issuance.
+pub fn mint_existing<T: Config>(
```

Comment:
I would keep the "into" part of this here, as it's quite different from `mint_creating`.
`mint_creating` = mint into an account that you create
`mint_into_existing` = mint into an existing account that you don't create

So to me `mint_into` and `mint_creating` would make more sense if you want to keep things short, or just keeping the `mint_into_existing` pattern from before